kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/Preconditions.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal;
+
+public class Preconditions {
```

Comment:
actually, could you please move this class to `io/delta/kernel/internal/util`? Also the utility class `io/delta/kernel/internal/util/InternalUtils.java` has similar methods. Could you remove them as well?

Reply:
Sure. I'll do it. Do you think I keep the name `Preconditions` or `PreconditionUtils` sounds better?

Reply:
> Sure. I'll do it. Do you think I keep the name `Preconditions` or `PreconditionUtils` sounds better?

Preconditions? The name is same as what guava or other libraries have. Easy to understand. Wdyt?

> 
> The `io/delta/kernel/internal/util` package only exist in the `kernel-api` module. Do you think it's better to move the `Preconditions` class to the `kernel-api` module or keep in the current one?

Lets move to `kernel-api` module under `io.delta.kernel.internal.util`. These methods are used in `kernel-api` as well. The defaults module can still use the APIs in `Preconditions` as `kernel-api` and `kernel-defaults` module are shipped together and easy to manage.

> 
> I was thinking about the class `DefaultKernelUtils`. Maybe it's a good idea to move it to the `io/delta/kernel/internal/util` package too. What do you think?

`DefaultKernelUtils` has few other methods which are only applicable to default ParquetHandler or other client implementations. Lets keep the `DefaultKernelUtils` minus the `checkArgument` methods remain there.

Reply:
> > Sure. I'll do it. Do you think I keep the name `Preconditions` or `PreconditionUtils` sounds better?
> 
> Preconditions? The name is same as what guava or other libraries have. Easy to understand. Wdyt?
> 

It sounds good to me. I raised the possibility in case you prefer to follow the naming of the other classes.
> > The `io/delta/kernel/internal/util` package only exist in the `kernel-api` module. Do you think it's better to move the `Preconditions` class to the `kernel-api` module or keep in the current one?
> 
> Lets move to `kernel-api` module under `io.delta.kernel.internal.util`. These methods are used in `kernel-api` as well. The defaults module can still use the APIs in `Preconditions` as `kernel-api` and `kernel-defaults` module are shipped together and easy to manage.
> 
I'll do it.
> > I was thinking about the class `DefaultKernelUtils`. Maybe it's a good idea to move it to the `io/delta/kernel/internal/util` package too. What do you think?
> 
> `DefaultKernelUtils` has few other methods which are only applicable to default ParquetHandler or other client implementations. Lets keep the `DefaultKernelUtils` minus the `checkArgument` methods remain there.
>
Great

Reply:
The class has been moved to the `kernel-api` module.